roles/delete_nodepools/tasks/main.yml

```diff
@@ -11,17 +11,19 @@
 
 - name: Use Terraform output in variables
   ansible.builtin.set_fact:
+
     k8s_cluster_id: "{{ terraform_run.outputs.cluster_id.value | default(omit) }}"
```

Comment:
default(omit) ensures that the variable 'k8s_cluster_id' is set to 'omit' when its value is not present or empty, instead of causing an error or generating an undesired value in your output